convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,243 @@
+# Copyright(C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+import os
+import shutil
+
+from convert2rhel import actions, grub, systeminfo
+from convert2rhel.grub import (
+    CENTOS_EFIDIR_CANONICAL_PATH,
+    DEFAULT_INSTALLED_EFIBIN_FILENAMES,
+    RHEL_EFIDIR_CANONICAL_PATH,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class NewDefaultEfiBin(actions.Action):
+    id = "NEW_DEFAULT_EFI_BIN"
+
```

Comment:
I wonder if there shouldn't be some dependency as `FIX_DEFAULT_KERNEL` (which is the last action from `preserve_only_rhel_kernel.py` if I'm right) and the EFI actions seems related to this

Reply:
We can add it

Reply:
@pr-watson @r0x0d Or... it depends. As I'm looking at the [original order](https://github.com/oamg/convert2rhel/blob/v1.7.1/convert2rhel/main.py#L299), it should be `transaction.py` > `preserve_only_rhel_kernel.py` > `list_non_red_hat_pgs_left.py` > `set_efi_config.py`. But the `list_non_red_hat_pgs_left.py` is missing its dependencies too, so we would have to add it (and it makes sense, or the `list_non_red_hat_pgs_left.py` could be possibly ran before the transaction which would be useless)

Reply:
If we decide to add the dependencies to `list_non_red_hat_pgs_left.py`, I can include it into the https://github.com/oamg/convert2rhel/pull/1341 where I'm touching this action

Reply:
Thanks for spotting the missing dependency, @hosekadam. Here's the PR that fixes the list_non_red_hat_pkgs_left.py: https://github.com/oamg/convert2rhel/pull/1342

Reply:
Okay so I should remove the new dependency here?

Reply:
So, as we discussed in the demo presentation about action framework a few weeks (months) ago, we don't strictly need every dependency to be in correct order. 

For example, the actions here doesn't need to have a dependency for `list_non_red_hat_pkgs_left`, as this is something that only makes sense for `transaction.py` actions. 

But regarding the original order, and you point, I think you're right, we want to add the dependencies at least for the preserve_only_rhel_kernel, and the preserve_only_rhel_kernel should have a dependency on transaction. 

I would vote for having this fixed in this PR, in that case, make the `preserve_only_rhel_kernel` a dependency of `set_efi_configuration`

Reply:
Alright so the dependency we have now will satisfy that